src/lib/mac/poly1305/poly1305.cpp

```diff
@@ -190,7 +190,11 @@ void Poly1305::final_result(byte out[])
    if(m_buf_pos != 0)
       {
       m_buf[m_buf_pos] = 1;
-      clear_mem(&m_buf[m_buf_pos+1], m_buf.size() - m_buf_pos - 1);
+      const auto len = m_buf.size() - m_buf_pos - 1;
```

Comment:
s/auto/size_t/ since that (I hope!) is what the compiler will deduce, and it makes for easier reading.